### PR TITLE
Allow .f90 files in kpp folder

### LINKED ISSUE #1105: Delete .gitignore in released file
I'm trying to add some personal functions to WRF-Chem.

So, I downloaded released [WRFV4.1.4](https://github.com/wrf-model/WRF/releases/tag/v4.1.4) and uploaded to my empty private repository.

Then, I cloned the repository and uploaded to the HPC.
But, I got this error when compiling WRF-Chem with KPP:
```
This is KPP-2.1.

KPP is parsing the equation file.
KPP was told to generate WRF conform code
KPP is computing Jacobian sparsity structure.
KPP is starting the code generation.
KPP is initializing the code generation.

 KPP is using the WRF conform integrator routine:
 /yin_raid/xin/tmp_lfr/WRF-4.1.4/chem/KPP/kpp/kpp-2.1/int/WRF_conform/rosenbrock

Fatal error : /yin_raid/xin/tmp_lfr/WRF-4.1.4/chem/KPP/kpp/kpp-2.1/int/WRF_conform/rosenbrock.f90: Can't read file
Program aborted
make: *** [cb05_sorg_aq_Integrator.f90] Error 3
grep: ../../../module_kpp_cb05_sorg_aq_Integr.F: No such file or directory
Makefile:50: warning: overriding commands for target `.F.o'
../../../../configure.wrf:379: warning: ignoring old commands for target `.F.o'
Makefile:58: warning: overriding commands for target `.c.o'
../../../../configure.wrf:408: warning: ignoring old commands for target `.c.o'
```

BTW, if I downloaded the official release directly on the HPC, it works well.

Actually, this issue is caused by the `*.f90` in the `.gitignore` ...
I think it's better to delete `.gitignore` in the released file.

### PR DESCRIPTION
**Pull request by zxdawn**
TYPE: bug fix

KEYWORDS: .gitignore, kpp, f90

DESCRIPTION OF CHANGES:
Problem:
The *.f90 files are part of .gitignore. This is not entirely correct. There are instances where the
*.f90 files should be tracked by git.

Solution:
Exclude the *.f90 files under the kpp directory from this overly broad identification of files. There
are other directories where this fix could also be applied (such as for DA).

ISSUE:
Fixes #1105 

LIST OF MODIFIED FILES:
M  .gitignore

TESTS CONDUCTED:
Compile and run WRF-Chem successfully

### CODE REVIEW

**Comment by davegill** [comment]
@zxdawn 
Xin,
In your pull request, would you more clearly state what the impact on the .f90 files due to your proposed changes to the .gitignore file. For example, explain the problem that WRF Chem KPP users see with the released code vs the cloned code, etc.

**Comment by davegill** [comment]
@zxdawn 
I just cloned the WRF repository and checked out the release-v4.1.4 branch. Then I pulled down the WRF release-v4.1.4 from the github download (the tar.gz file). 

I must be misunderstanding. There are identical *.f90 files in both directory structures. What is your PR addressing?

**Comment by zxdawn** [comment]
@davegill Sorry for the misleading. This is for a special case.
Let's assume I want to add a personal module to WRFV4.1.4, but I don't want to share with others currently.

My steps are shown below:
1. Create an empty private repository.
2. Download the WRFV4.1.4 release and extract it.
3. Using `git add -A`, `git commit -m "*****"` and `git push` to upload online.

Because the extracted WRF codes are all new to the empty repository, the `.gitignore` will cause the loss of `.f90` files in kpp folder. But, for your case, `.f90` files are old files which aren't affected by `.gitignore`.

File: .gitignore [hunk 1]
```diff
@@ -16,3 +16,4 @@
 configure.wrf*
 *.backup
 *.f90
+!chem/KPP/kpp/kpp-2.1/**/*.f90
```
Comment (davegill) [inline .gitignore]: @zxdawn 
Please explain the semantics and syntax in this file. For example, 
1. what is the impact of two rules governing the *.f90 files
2. what do two `**` together signify

Comment (zxdawn) [reply .gitignore]: @davegill `*.f90` means ignore all `.f90` files. After `!chem/KPP/kpp/kpp-2.1/**/*.f90` is added, .f90 files under all subdirectories of `chem/KPP/kpp/kpp-2.1/` will be kept.

I have checked all names of .f90 files under these subdirectories after the compilation of WRF-Chem.

They are as same as these stored in GitHub now. So, this method could keep necessary .f90 files for KPP and ignore many .f90 files generated after compilation of WRF.

**Comment by davegill** [comment]
@zxdawn 
Xin,
Instead of modifying the top-level .gitignore file, move these modifications to a new .gitignore file that is in the kpp subdirectory.

**Comment by zxdawn** [comment]
@davegill Moved.

**Comment by davegill** [comment]
@zxdawn @jordanschnell @kayeekayee 
Xin,
Yes, this is a much better idea

Jordan and Ka Yee,
Do you see any downside to this introduction of a .gitignore file in the kpp directory?

**Pull request merged**